rllib/algorithms/impala/impala.py

```diff
@@ -664,7 +665,7 @@ def training_step(self) -> ResultDict:
         with self._timers[SYNCH_WORKER_WEIGHTS_TIMER]:
             if self.config._enable_learner_api:
                 if train_results:
-                    pids = list(train_results["learner"]["loss"].keys())
+                    pids = list(set(train_results.keys()) - {ALL_MODULES})
```

Comment:
this feels like dark magic, is there any test or assert or something to make sure that nobody would add random additional keys to train_results?

Reply:
good point. I'll assert something in the learner.